### ISSUE TITLE
`restrict_to_interval` does not work for individual trials

### ISSUE THREAD

**Issue by raeedcho** [mattperich/PyalData#112]
Trying to use `restrict_to_interval` on an individual trial (which is a Pandas Series, rather than a DataFrame) doesn't seem to work (many of the functions used by `restrict_to_interval` require a DataFrame input).

In fact, it seems that perhaps many functions would fail if given a single trial (e.g. `get_time_varying_fields`).

A possible solution might be to put in explicit checks in functions for single trial input instead of a DataFrame. I'm not sure if that would need to be done in every function that assumes a DataFrame, or if there's a more elegant solution available.

**Comment by bagibence** [comment]
Yes, one of the reasons for this is that currently anything that creates a copy of the input will fail if the first positional argument or the "trial_data" keyword argument is not a DataFrame because the `@utils.copy_td` decorator throws an error in this case.

I guess turning the Series into a DataFrame with one row might fix some of these. The problem with that is when applying functions to a Series, you expect them to return a Series (and not a DataFrame), right?

**Comment by raeedcho** [comment]
Hmm, good point. I think the best solution would be to explicitly turn my individual trial from a Series into a one-row DataFrame before running functions on it meant for multi-trial data and leave it at that. I probably should have thought of this sooner, except I'm very new to working with pandas, so I don't know its ins and outs yet 😁 

I'm happy to close this issue with that solution though, if you agree.

**Comment by bagibence** [comment]
Don't close it just yet :) It's a good point (and issue) and the code should work on `Series` objects as well.
I was thinking that probably the easiest way of solving this is that if the functions get a `Series` as the input, they convert it to a `DataFrame`, do their thing, then -- maybe only some of them -- convert it back to a `Series` (instead of returning a one-row `DataFrame`. I guess this might be done within `utils.copy_td` or with an additional decorator?

What I would suggest is that first we organize the code a bit and move everything that is now just dumped into `utils` and `tools` into smaller modules, then return to this. Doing that might make it a bit more clear which parts of the codebase should be modified to fix this. What do you think?
